x-pack/plugins/ml/server/models/annotation_service/annotation.ts

```diff
@@ -332,6 +342,79 @@ export function annotationProvider({ asInternalUser }: IScopedClusterClient) {
     }
   }
 
+  /**
+   * Fetches the latest delayed data annotation per job.
+   * @param jobIds
+   */
+  async function getDelayedDataAnnotations({
+    jobIds,
+    earliestMs,
```

Comment:
There should be a "lastCheckMs" that filters out anything where the `modified_time` is less than it.